exporter/exporter.go

```diff
@@ -87,15 +89,33 @@ func New(opts *Opts) (*Exporter, error) {
 		logger:           opts.Logger,
 		opts:             opts,
 		webListenAddress: opts.WebListenAddress,
-		topologyInfo:     ti,
+		lock:             &sync.Mutex{},
 	}
 
 	return exp, nil
 }
 
+//nolint:funlen
 func (e *Exporter) makeRegistry(ctx context.Context, client *mongo.Client) *prometheus.Registry {
-	// TODO: use NewPedanticRegistry when mongodb_exporter code fulfils its requirements (https://jira.percona.com/browse/PMM-6630).
-	registry := prometheus.NewRegistry()
+	registry := prometheus.NewPedanticRegistry()
+
+	e.lock.Lock()
+	if e.topologyInfo == nil {
+		var err error
+
+		if e.topologyInfo, err = newTopologyInfo(ctx, client); err != nil {
```

Comment:
e.topologyInfo` can be local variable as we not use it outside of the current method.
In this case, `e.lock` will obsolete as well.